### DISCUSSION TITLE
Implementation guidelines for supporting updates in CR

### DISCUSSION THREAD

**Discussion by juangon** [operator-framework/java-operator-sdk#1609] [Q&A] [answered]
Hi! I would like to implement the Operator I implemented in update modes (instead of just create and deletion).

AFAIK all resource updates should go into Reconciler's reconcile method, but it doesn't look clear to me how to update Dependent Resources etc.

Can someone guide me? Thanks very much!

**Comment by metacosm** [comment, marked as answer]
Your `DependentResource` implementations need to implement the `Updater` interface and they will be automatically updated when needed by JOSDK.

**Reply by juangon** [reply]
Ok thanks for the answer. Apart from that, the remaining logic would go into Reconciler's reconcile method, right? For example if I want to restart a deployment when some dependent configmap had been updated

**Reply by metacosm** [reply]
I guess it depends, it could also be done in the configmap dependent resource… It depends where it makes more sense. This could also be modelled using a workflow if you can define a condition that would prevent the deployment workflow to proceed further until the configmap has been updated.

**Reply by juangon** [reply]
> Your `DependentResource` implementations need to implement the `Updater` interface and they will be automatically updated when needed by JOSDK.

It seems it doesn't work as it should @metacosm . So the first level Dependent Resource update method is invoked, but it doesn't kicks in the second level dependent resources. It even doesn't invoke the "desired" method from those second level dependent resources. Can you give a hint on what to check please?

**Reply by juangon** [reply]
> > Your `DependentResource` implementations need to implement the `Updater` interface and they will be automatically updated when needed by JOSDK.
> 
> It seems it doesn't work as it should @metacosm . So the first level Dependent Resource update method is invoked, but it doesn't kicks in the second level dependent resources. It even doesn't invoke the "desired" method from those second level dependent resources. Can you give a hint on what to check please?

It seems that reconciler isn't being invoked at that level. Do you know why that could happen?

Thanks very much!

**Reply by juangon** [reply]
Ok so I upgraded to the new Quarkus-operator-sdk 5.0.0.Beta2, which has JOSDK 4.1.1 with this fix: https://github.com/java-operator-sdk/java-operator-sdk/issues/1589

So I removed all the update methods from Dependents, and now seems to work!! Just have to touch some reconcilers to take into account the updates, but that's another story
